src/data/decimal/decimalFn.ts

```diff
@@ -0,0 +1,72 @@
+import { assert } from '../../utils/assert'
+import { PLDecimal } from './PLDecimal'
+
+export function plDecimal(strValue: string): PLDecimal {
+  const decimalObj = parseNumString(strValue)
+  return new PLDecimal(decimalObj.intValue, decimalObj.decimals)
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertNumeric(strValue: any): any {
+  return assert(strValue === '' || isNaN(strValue), `Invalid decimal number: "${strValue}"`)
+}
+
+export function isScientific(strValue: string): boolean {
+  const parts = strValue.split(/[eE]/)
+  if (parts.length === 2 && assertNumeric(parts[0]) && assertNumeric(parts[1])) {
+    const intValueRounded = Math.abs(parseInt(parts[0]))
+    const exponential = parseFloat(parts[1])
+    return intValueRounded >= 1 && intValueRounded <= 9 && exponential === parseInt(parts[1])
+  } else {
+    return false
+  }
+}
+
+export function parseScientificString(strValue: string): any {
```

Comment:
Looks like you return with `{intValue, decimals}` objects few times, so you can create a helper interface for that fter the imports:

```
interface DecimalResult {
   intValue: number, 
   decimals: number
}

then you can use it instead `any`

Reply:
done